cmd/abapEnvironmentRunAUnitTest.go

```diff
@@ -106,40 +104,49 @@ func runAbapEnvironmentRunAUnitTest(config *abapEnvironmentRunAUnitTestOptions,
 }
 
 func triggerAUnitrun(config abapEnvironmentRunAUnitTestOptions, details abaputils.ConnectionDetailsHTTP, client piperhttp.Sender) (*http.Response, error) {
-	var aUnitConfigYamlFile []byte
+
 	abapEndpoint := details.URL
-	filelocation, err := filepath.Glob(config.AUnitConfig)
-	//Parse YAML AUnit run configuration as body for AUnit run trigger
-	if err == nil {
-		filename, err := filepath.Abs(filelocation[0])
-		if err == nil {
-			aUnitConfigYamlFile, err = ioutil.ReadFile(filename)
-		}
-	}
-	var AUnitConfig AUnitConfig
-	if err == nil {
-		var result []byte
-		result, err = yaml.YAMLToJSON(aUnitConfigYamlFile)
-		json.Unmarshal(result, &AUnitConfig)
-	}
-	var metadataString, optionsString, objectSetString string
-	if err == nil {
-		metadataString, optionsString, objectSetString, err = buildAUnitTestBody(AUnitConfig)
+	bodyString, err := buildAUnitRequestBody(config)
+	if err != nil {
+		return nil, err
 	}
 
 	//Trigger AUnit run
 	var resp *http.Response
-	var bodyString = `<?xml version="1.0" encoding="UTF-8"?>` + metadataString + optionsString + objectSetString
+
 	var body = []byte(bodyString)
-	if err == nil {
-		log.Entry().Debugf("Request Body: %s", bodyString)
-		details.URL = abapEndpoint + "/sap/bc/adt/api/abapunit/runs"
-		resp, err = runAUnit("POST", details, body, client)
-	}
-	if err != nil {
-		return resp, fmt.Errorf("Triggering AUnit test run failed: %w", err)
+	log.Entry().Debugf("Request Body: %s", bodyString)
+	details.URL = abapEndpoint + "/sap/bc/adt/api/abapunit/runs"
+	resp, err = runAUnit("POST", details, body, client)
+	return resp, err
+}
+
+func resolveAUnitConfiguration(config abapEnvironmentRunAUnitTestOptions) (aUnitConfig AUnitConfig, err error) {
```

Comment:
Same here, I think we should do it universally for the `ATC` and `AUnit` and resolve the configuration with a helper method outside of the step itself. Since both ATC and AUnit need to be provided with the list of software components in case of the `Repositories` flag I think this is feasible

Reply:
The mapping to the resulting data type is different - I don't think that this would be beneficial. The tool independent part is already reusable